File: cpe49題/10922 2 the 9s.cpp
#include <iostream>
using namespace std;

int result;

void degreeFun(string s){
	int tmp1,sum=0;
	for(int i=0;i<s.length();i++){
		tmp1=s[i]-'0';
		sum+=tmp1;
	}
	//cout<<"sum:"<<sum<<endl;
	if(sum%9==0){
		if(sum>9){
			result++;
			s=to_string(sum);
			degreeFun(s);
		}
	}else{
		result=0;
	}
}

int main() {
	string s;
	while(cin>>s){
		result=1;
		if(s=="0") break;
		degreeFun(s);
		if(result!=0){
			cout<<s<<" is a multiple of 9 and has 9-degree "<<result
<<"."<<endl;
		}else{
			cout<<s<<" is not a multiple of 9."<<endl;
		}		
	}
}

File: cpe49題/10922 2 the 9s.py
while True:
	s=input()
	if s=="0":
		break
	
	numList=[]
	for c in s:
		numList.append(int(c))
	
	if sum(numList)%9!=0:
		print(f"{s} is not a multiple of 9.")
		continue
		
	num=sum(numList)
	degree=1
	if num==9:
		degree=1
	else:
		while num>9:
			tmp=str(num)
			total=0
			for i in tmp:
				total+=int(i)
			num=total
			degree+=1
	print(f"{s} is a multiple of 9 and has 9-degree {degree}.")
